### PR TITLE
[frogend] (Mastodon) domain block CSV import

### PR DESCRIPTION
**Pull request by f0x52**
# Description
Adds CSV import and export to the bulk import/export interface, tested to be compatible with Mastodon too (glitch-soc).
Generally cleans up the code structure around it a bit too, both on the page itself and the exportType based branching in query.

Also added some information text about the page, and a table comparing the different export formats. Still not sure how well it fits there, might just have to move that to the docs, or a separate page?

## Checklist

Please put an x inside each checkbox to indicate that you've read and followed it: `[ ]` -> `[x]`

If this is a documentation change, only the first checkbox must be filled (you can delete the others if you want).

- [x] I/we have read the [GoToSocial contribution guidelines](https://github.com/superseriousbusiness/gotosocial/blob/main/CONTRIBUTING.md).
- [x] I/we have discussed the proposed changes already, either in an issue on the repository, or in the Matrix chat.
- [x] I/we have performed a self-review of added code.
- [x] I/we have written code that is legible and maintainable by others.
- [x] I/we have commented the added code, particularly in hard-to-understand areas.
- [x] I/we have made any necessary changes to documentation.
- [x] I/we have added tests that cover new code.
- [x] I/we have run tests and they pass locally with the changes.
- [x] I/we have run `go fmt ./...` and `golangci-lint run`.

### CODE REVIEW

**Comment by tsmethurst** [comment]
This looks really good to me :) Just wondering about what we discussed briefly in matrix: domains that are wildcarded by starting with `*.`. Should we trim that off before it gets sent to gts, or should the server do it?

**Comment by f0x52** [comment]
hmm I feel like the server should reject them if it doesn't handle it for you, because as a literal * it's never a valid domain. If we don't want to handle them in the backend I can update the client validation to match, and possibly add the trimming there

I'm actually not sure if Mastodon even handles those wildcards either..

**Comment by tsmethurst** [comment]
I can add something to trim it in the server side, then it'll work via the API too and not just the settings panel: I needed a small thing to work on this afternoon anyway :)

File: web/source/settings/admin/federation/import-export.js [hunk 1]
```diff
@@ -94,49 +95,72 @@ module.exports = function ImportExport() {
...
 				</h1>
-				<FormWithData
-					dataQuery={query.useInstanceBlocksQuery}
-					DataForm={ImportList}
-					list={parseResult.data}
-				/>
+				<div className="without-border">
+					<FormWithData
+						dataQuery={query.useInstanceBlocksQuery}
+						DataForm={ImportList}
```
Comment (tsmethurst) [inline web/source/settings/admin/federation/import-export.js:101]: very small nitpick, feel free to ignore: is there a reason for the different casing of DataForm?

File: web/source/settings/lib/domain-block.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+			Wildcard prefix *. can be stripped since it's equivalent to not having it,
+			but wildcard anywhere else in the domain is not handled by the backend so it's invalid.
+		*/
+		wildcard: false,
+		allowUnicode: true
+	});
+}
+
+/* 
+	Still can't think of a better function name for this,
```
Comment (tsmethurst) [inline web/source/settings/lib/domain-block.js:36]: `canBlockHigher`? i can't really think of one either; think it's fine as-is though

**Review by tsmethurst — COMMENTED** [review]
This looks good to me so far :)

File: web/source/settings/admin/federation/import-export.js (hunk 1 above)
Comment (f0x52) [reply web/source/settings/admin/federation/import-export.js:101]: because it's a React component, which are recommended as PascalCase for some reason

**Pull request merged**